apps/wallet-dashboard/hooks/useGetStardustMigratableObjects.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+import { useGetCurrentEpochStartTimestamp } from '@/hooks';
+import { groupStardustObjectsByMigrationStatus } from '@/lib/utils';
+import { useFeature } from '@growthbook/growthbook-react';
+import {
+    Feature,
+    STARDUST_BASIC_OUTPUT_TYPE,
+    STARDUST_NFT_OUTPUT_TYPE,
+    useGetAllOwnedObjects,
+} from '@iota/core';
+import { IotaObjectData } from '@iota/iota-sdk/client';
+
+export function useGetStardustMigratableObjects(address: string): {
+    migratableBasicOutputs: IotaObjectData[];
+    migratableNftOutputs: IotaObjectData[];
+} {
+    const { data: currentEpochMs } = useGetCurrentEpochStartTimestamp();
+    const { data: basicOutputObjects } = useGetAllOwnedObjects(address, {
+        StructType: STARDUST_BASIC_OUTPUT_TYPE,
+    });
+    const { data: nftOutputObjects } = useGetAllOwnedObjects(address, {
+        StructType: STARDUST_NFT_OUTPUT_TYPE,
+    });
+
+    const stardustMigrationEnabled = useFeature<boolean>(Feature.StardustMigration).value;
```

Comment:
Do we need this feature check inside of the hook, Especially if we have a same check in the home/page.tsx?
Also maybe the hook shouldnt have the dependency of the StardustMigration feature which is kind of "hidden" inside of the hook and  its not immediately visible that the its needed so the hook returns the results. wdyt?

Reply:
I added the feature check inside the hook because @cpl121 recommended it, the idea is that if the feature is disabled, the hook in home would still trigger unnecessary fetches, which could slow things down since home already fetches coins, activities...

Reply:
never mind, the grid needs that check :see_no_evil:

Reply:
I think it's better to share the `stardustMigrationEnabled` by prop to the hook. My point is that if the Feature is disabled, it doesnt make sense to fetch all the migration objects in the home
I will put another comment with the proposed solution